tasks/main.yml

```diff
@@ -1,6 +1,11 @@
 ---
-- name: add the OS specific variables
-  include_vars: '{{ ansible_os_family }}.yml'
+- name: Set OS dependent variables
+  include_vars: "{{ item }}"
+  with_first_found:
+   - '{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml'
+   - '{{ ansible_distribution }}.yml'
+   - '{{ ansible_os_family }}-{{ ansible_distribution_major_version }}.yml'
+   - '{{ ansible_os_family }}.yml'
```

Comment:
Maybe include `{{ ansible_os_family }}.yml` always a first task and then have a second task which does the `with_first_found`? This way redundancy could be decreased.